### ISSUE TITLE
How to get the nth address in a network?

### ISSUE THREAD

**Issue by fadedbee** [stbuehler/rust-cidr#17]
I'm currently using:
```
fn nth(base: Ipv4Inet, n: usize) -> Option<Ipv4Inet> {
    let mut a = base;

    // FIXME: This is pointlessly inefficient.
    for _ in 0..n {
        a = a.next()?;
    }

    Some(a)
}
```

How should I do this properly, without iteration?

**Comment by stbuehler** [comment]
Hi.

So the [`next()` implementation](https://github.com/stbuehler/rust-cidr/blob/5fdc64ecd4aff824b728b78353e12dd190195290/src/internal_traits.rs#L89) actually simply uses the `u32` representation (or `u128` for IPv6); the code is a bit complicated because I wanted it to be `const`, and I made it somewhat generic so it could be used to implement something like your `nth` method too - the underlying methods are actually already there, just commented.

I think I didn't expose them yet because I wasn't sure how a proper API for those should look like.

Something like this might do it without `const`:
```rust
fn nth(base: Ipv4Inet, n: u32) -> Option<Ipv4Inet> {
    let result = Ipv4Inet::new(
        u32::from(base.address()).checked_add(n)?.into(),
        base.network_length(),
    ).expect("network length was valid before");
    if result.network() == base.network() {
        Some(result)
    } else {
        None
    }
}
```

**Comment by stbuehler** [comment]
Added new methods in 6f506ca7d12f7798eb4aa662b64c8b0497e8957b and published in [0.2.3](https://github.com/stbuehler/rust-cidr/releases/tag/cidr-0.2.3)
